jax/_src/lib/__init__.py

```diff
@@ -41,33 +42,46 @@
     'https://github.com/google/jax#installation for installation instructions.'
     ) from err
 
+import jax.version
 from jax.version import _minimum_jaxlib_version as _minimum_jaxlib_version_str
 try:
-  import jaxlib.version as jaxlib_version
+  import jaxlib.version
 except Exception as err:
   # jaxlib is too old to have version number.
   msg = f'This version of jax requires jaxlib version >= {_minimum_jaxlib_version_str}.'
   raise ImportError(msg) from err
 
-version = tuple(int(x) for x in jaxlib_version.__version__.split('.'))
-_minimum_jaxlib_version = tuple(int(x) for x in _minimum_jaxlib_version_str.split('.'))
 
 # Check the jaxlib version before importing anything else from jaxlib.
 def _check_jaxlib_version():
-  if version < _minimum_jaxlib_version:
-    msg = (f'jaxlib is version {jaxlib_version.__version__}, '
-           f'but this version of jax requires version {_minimum_jaxlib_version_str}.')
-
-    if version == (0, 1, 23):
-      msg += ('\n\nA common cause of this error is that you installed jaxlib '
-              'using pip, but your version of pip is too old to support '
-              'manylinux2010 wheels. Try running:\n\n'
-              'pip install --upgrade pip\n'
-              'pip install --upgrade jax jaxlib\n')
-    raise ValueError(msg)
+  version_regex = re.compile(r"[0-9]+\.[0-9]+(?:\.[0-9]+)*")
+  def _parse_version(v: str) -> Tuple[int, ...]:
+    m = version_regex.match(v)
+    if m is None:
+      raise ValueError(f"Unable to parse jaxlib version '{v}'")
+    return tuple(int(x) for x in m.group(0).split('.'))
+
+  _jax_version = _parse_version(jax.version.__version__)
+  _minimum_jaxlib_version = _parse_version(jax.version._minimum_jaxlib_version)
+  _jaxlib_version = _parse_version(jaxlib.version.__version__)
+
+  if _jaxlib_version < _minimum_jaxlib_version:
+    msg = (f'jaxlib is version {jaxlib.version.__version__}, but this version '
+           f'of jax requires version >= {jax.version._minimum_jaxlib_version}.')
+    raise RuntimeError(msg)
+
+  if _jaxlib_version[:2] > _jax_version[:2]:
```

Comment:
Are there tests for this code? Is it possible to add them?

Reply:
No, I don't see how you could without doing something like monkey patching the jax versions. And we wouldn't be able to run those tests in pytest because the imports are shared across tests.

Reply:
There you go I added a test.